### PR TITLE
Avoid Multiple Notifications on Expired Session

### PR DESCRIPTION
**Pull request by djhi**

### CODE REVIEW

File: packages/ra-core/src/auth/useLogoutIfAccessDenied.ts [hunk 1]
```diff
@@ -40,21 +42,32 @@ const useLogoutIfAccessDenied = (): LogoutIfAccessDenied => {
...
-                    notify('ra.notification.logged_out', 'warning');
-                    return true;
-                }),
+        (error?: any) => {
+            // Sometimes, a component might trigger multiple simultaneous
+            // dataProvider calls which all fail and call this function.
+            // To avoid having multiple notifications, we first verify if
+            // a checkError promise is already ongoing
+            if (!authCheckPromise) {
+                authCheckPromise = authProvider
```
Comment (fzaninotto) [inline packages/ra-core/src/auth/useLogoutIfAccessDenied.ts:51]: why do you store the entire Promise instead of just a boolean?

Comment (djhi) [reply packages/ra-core/src/auth/useLogoutIfAccessDenied.ts:51]: Because this way, anyone interested by its result will receive it ?

Comment (fzaninotto) [reply packages/ra-core/src/auth/useLogoutIfAccessDenied.ts:51]: Hm, you're exposing a private API publicly without a real use case... I'm not comfortable with it.

Comment (djhi) [reply packages/ra-core/src/auth/useLogoutIfAccessDenied.ts:51]: We were returning it before already. Nothing has changed. We just don't allow multiple simultaneous calls, that's all

Comment (fzaninotto) [reply packages/ra-core/src/auth/useLogoutIfAccessDenied.ts:51]: Gotcha, thanks for the explanation.

**Pull request merged**